tests/unit/test_charm.py

```diff
@@ -1,18 +1,128 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+from typing import Dict, List, Literal
+from unittest.mock import Mock, patch
+
 import ops
 import ops.testing
 import pytest
+import requests
 from charm import GocertCharm
-from scenario import Context, Event, State
+from scenario import Container, Context, Event, Network, State, Storage
+
+# https://res.cloudinary.com/canonical/image/fetch/f_auto,q_auto/https://discourse-charmhub-io.s3.eu-west-2.amazonaws.com/original/2X/4/4ac42dc8a238a003c7d56fe282246ca102dd594f.png
 
 
 class TestCharm:
-    @pytest.fixture(scope="function", autouse=True)
+    @pytest.fixture(scope="function")
     def context(self):
         yield Context(GocertCharm)
 
-    def test_start_charm(self, context):
-        out = context.run(Event("start"), State())
-        assert out.unit_status == ops.ActiveStatus()
+    @pytest.mark.parametrize(
```

Comment:
This unit test is huge , are you trying to merge all tests into 1. It feels like we went too far here  :laughing:

Reply:
There is only 1 public function :D.

I parameterized all of the tests and moved this to a separate PR.